src/sentry/search/events/builder/metrics.py

```diff
@@ -134,7 +135,13 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
             if self.params.environments:
                 environment = self.params.environments[0].name
 
-            return OnDemandMetricSpec(field, self.query, environment, groupby_columns)
+            return OnDemandMetricSpec(
+                field=field,
+                query=self.query,
+                environment=environment,
+                groupbys=groupby_columns,
+                spec_type=self.builder_config.on_demand_metrics_type or MetricSpecType.SIMPLE_QUERY,
```

Comment:
I see what you're saying but that could cause a fork in behaviour for alerts that I'm trying to avoid in this PR, how about instead we throw an exception if it's empty and we can see if this was somehow not defined and go fix it.

Reply:
Actually nvm it's in the querying, I'm less concerned